jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2488,35 +2489,35 @@ code_blockt java_bytecode_convert_methodt::convert_newarray(
   const exprt::operandst &op,
   exprt::operandst &results)
 {
-  char element_type;
-
-  if(statement == "newarray")
-  {
-    irep_idt id = arg0.type().id();
-
-    if(id == ID_bool)
-      element_type = 'z';
-    else if(id == ID_char)
-      element_type = 'c';
-    else if(id == ID_float)
-      element_type = 'f';
-    else if(id == ID_double)
-      element_type = 'd';
-    else if(id == ID_byte)
-      element_type = 'b';
-    else if(id == ID_short)
-      element_type = 's';
-    else if(id == ID_int)
-      element_type = 'i';
-    else if(id == ID_long)
-      element_type = 'j';
+  java_reference_typet ref_type = [&]() {
+    if(statement == "newarray")
+    {
+      irep_idt id = arg0.type().id();
+
+      char element_type;
+      if(id == ID_bool)
+        element_type = 'z';
+      else if(id == ID_char)
+        element_type = 'c';
+      else if(id == ID_float)
+        element_type = 'f';
+      else if(id == ID_double)
+        element_type = 'd';
+      else if(id == ID_byte)
+        element_type = 'b';
+      else if(id == ID_short)
+        element_type = 's';
+      else if(id == ID_int)
+        element_type = 'i';
+      else if(id == ID_long)
+        element_type = 'j';
+      else
+        element_type = '?';
+      return java_array_type(element_type);
+    }
     else
-      element_type = '?';
-  }
-  else
-    element_type = 'a';
-
-  const reference_typet ref_type = java_array_type(element_type);
+      return java_reference_array_type(to_struct_tag_type(arg0.type()));
+  }();
```

Comment:
The change is because `java_reference_typet` doesn't have a nullary constructor, so we can't use the old structure of a default-init'd variable which is then assigned one way or the other

Reply:
Sure - but you can do, refactor old code to use immmediately invoked lambda, then add the extra condition?